### PR TITLE
cipher: 2.0.0 -> 2.5.0

### PR DESCRIPTION
**Pull request by xiorcale**
###### Motivation for this change

Version update and `postPatch` cleaning since https://github.com/arshubham/cipher/pull/44 is part of the new release.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/misc/cipher/default.nix [hunk 1]
```diff
@@ -42,10 +41,8 @@ stdenv.mkDerivation rec {
   ];
 
   postPatch = ''
-  	substituteInPlace data/com.github.arshubham.cipher.desktop.in \
-  		--replace xdg-open ${xdg_utils}/bin/xdg-open
```
Comment (worldofpeace) [inline pkgs/applications/misc/cipher/default.nix]: I still think it's a good idea to do this with `gio`.

```
substituteInPlace data/com.github.arshubham.cipher.desktop.in \
  		--replace "gio" "${glib.bin}/bin/gio"

Comment (kjuvi) [reply pkgs/applications/misc/cipher/default.nix]: Hum, I thought the PR was to get rid of the substitution, but it was to remove a dependency, my bad.

File: pkgs/applications/misc/cipher/default.nix [hunk 2]
```diff
@@ -42,10 +41,10 @@ stdenv.mkDerivation rec {
...
-    chmod +x post_install.py
-    patchShebangs post_install.py
+    substituteInPlace data/com.github.arshubham.cipher.desktop.in \
+                    --replace "gio" "${glib.bin}/bin/gio"
```
Comment (worldofpeace) [inline pkgs/applications/misc/cipher/default.nix]: Think the indentation here is a bit deep, can we use two spaces? Otherwise LGTM.

Comment (kjuvi) [reply pkgs/applications/misc/cipher/default.nix]: Guilty, I copy/paste you're suggestion :sweat_smile:

**Review by worldofpeace — APPROVED** [review]
Works for me :+1: Thank you @kjuvi .

**Pull request merged**